lib/dor/services/public_xml_service.rb

```diff
@@ -11,42 +11,94 @@ def to_xml
       pub['id'] = object.pid
       pub['published'] = Time.now.utc.xmlschema
       pub['publishVersion'] = 'dor-services/' + Dor::VERSION
-      release_xml = Nokogiri(generate_release_xml)
 
-      im = object.datastreams['identityMetadata'].ng_xml.clone
-      im.search('//release').each(&:remove) # remove any <release> tags from public xml which have full history
+      pub.add_child(public_identity_metadata.root) # add in modified identityMetadata datastream
+      pub.add_child(public_content_metadata.root) if public_content_metadata.xpath('//resource').any?
+      pub.add_child(public_rights_metadata.root)
 
-      pub.add_child(im.root) # add in modified identityMetadata datastream
-      public_content_metadata = object.datastreams['contentMetadata'].public_xml
-      pub.add_child(public_content_metadata.root.clone) if public_content_metadata.xpath('//resource').any?
-      pub.add_child(object.datastreams['rightsMetadata'].ng_xml.root.clone)
-
-      rels = object.public_relationships.root
-      pub.add_child(rels.clone) unless rels.nil? # TODO: Should never be nil in practice; working around an ActiveFedora quirk for testing
-      pub.add_child(object.generate_dublin_core.root.clone)
-      pub.add_child(Nokogiri::XML(object.generate_public_desc_md).root.clone)
-      pub.add_child(release_xml.root.clone) unless release_xml.xpath('//release').children.size == 0 # If there are no release_tags, this prevents an empty <releaseData/> from being added
+      pub.add_child(public_relationships.root) unless public_relationships.nil? # TODO: Should never be nil in practice; working around an ActiveFedora quirk for testing
+      pub.add_child(object.generate_dublin_core.root)
```

Comment:
i had left my question in the first commit, the line got moved in the second commit, so github shows the comment as being "outdated", but question still applies here, i think.